Modules/_sqlite/connection.c

```diff
@@ -263,6 +263,12 @@ void pysqlite_connection_dealloc(pysqlite_Connection* self)
  */
 int pysqlite_connection_register_cursor(pysqlite_Connection* connection, PyObject* cursor)
 {
+    if (!connection || !connection->cursors) {
+        PyErr_Format(PyExc_RuntimeError,
```

Comment:
`pysqlite_check_connection()` is used by most of the methods of `Connection` to verify that the `Connection` object is fine, and it raises a `ProgrammingError` otherwise, so for the uniformity of the code, i think `ProgrammingError` is more appropriate here.